ACHNBrowserUI/ACHNBrowserUI/packages/Backend/Sources/Backend/models/DailyTasks.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  File.swift
+//  
+//
+//  Created by Jan on 19.05.20.
+//
+
+import Foundation
+
+public struct DailyTasks: Codable {
+    
+    public struct Task: Codable {
+        public let hasProgress: Bool
+        public let maxProgress: Int
+        public var curProgress = 0 {
+            didSet {
+                if curProgress > maxProgress {
+                    curProgress = 0
+                }
+            }
+        }
+        
+        public init(hasProgress: Bool, maxProgress: Int) {
+            self.hasProgress = hasProgress
+            self.maxProgress = maxProgress
+        }
+        
+        public mutating func increaseProgress() {
+            curProgress += 1
+            
+            if curProgress > maxProgress {
+                curProgress = 1
+            }
+        }
+    }
+
+    public var lastUpdate = Date.init()
+    public var tasks = [
+        "rocks": Task(hasProgress: true, maxProgress: 8),
```

Comment:
I would have the name as an Enum: String, prettier :)

Reply:
And with the Enum you can index it in the dictionary by the raw value `[Tasks.Rock.rawValue: Task`] ...